Signal/src/call/CallService.swift

```diff
@@ -388,11 +389,17 @@ protocol CallServiceObserver: class {
     public func handleMissedCall(_ call: SignalCall, thread: TSContactThread) {
         AssertIsOnMainThread()
         // Insert missed call record
-        let callRecord = TSCall(timestamp: NSDate.ows_millisecondTimeStamp(),
-                                withCallNumber: thread.contactIdentifier(),
-                                callType: RPRecentCallTypeMissed,
-                                in: thread)
-        callRecord.save()
+        if call.callRecord == nil {
+            call.callRecord = TSCall(timestamp: NSDate.ows_millisecondTimeStamp(),
+                                     withCallNumber: thread.contactIdentifier(),
+                                     callType: RPRecentCallTypeMissed,
+                                     in: thread)
+        } else if (call.callRecord!.callType == RPRecentCallTypeIncoming) {
+            call.callRecord!.updateCallType(RPRecentCallTypeMissed)
+        }
+
+        assert(call.callRecord != nil)
```

Comment:
:+1: I've been doing conservative things like this too. It really shouldn't be nil, but don't explode about it.